test/test_constraints.py

```diff
@@ -0,0 +1,63 @@
+import unittest
+import pytest
+import numpy as np
+from sklearn.utils import shuffle
+from metric_learn.constraints import Constraints
+
+SEED = 42
+
+
+def gen_labels_for_chunks(num_chunks, chunk_size,
+                          n_classes=10, n_unknown_labels=5):
+  """Generates num_chunks*chunk_size labels that split in num_chunks chunks,
+  that are homogeneous in the label."""
+  assert min(num_chunks, chunk_size) > 0
+  classes = shuffle(np.arange(n_classes), random_state=SEED)
+  n_per_class = chunk_size * (num_chunks // n_classes)
+  n_maj_class = chunk_size * num_chunks - n_per_class * (n_classes - 1)
+
+  first_labels = classes[0] * np.ones(n_maj_class, dtype=int)
+  remaining_labels = np.concatenate([k * np.ones(n_per_class, dtype=int)
+                                     for k in classes[1:]])
+  unknown_labels = -1 * np.ones(n_unknown_labels, dtype=int)
+
+  labels = np.concatenate([first_labels, remaining_labels, unknown_labels])
+  return shuffle(labels, random_state=SEED)
+
+
+class TestConstraints(unittest.TestCase):
+  def test_exact_num_points_for_chunks(self):
+    """Checks that the chunk generation works well with just enough points."""
+    for num_chunks, chunk_size in [(5, 10), (10, 50)]:
+      labels = gen_labels_for_chunks(num_chunks, chunk_size)
+
+      constraints = Constraints(labels)
+      chunks = constraints.chunks(num_chunks=num_chunks, chunk_size=chunk_size,
+                                  random_state=SEED)
+
+      chunk_no, size_each_chunk = np.unique(chunks, return_counts=True)
+
+      self.assertTrue(np.all(size_each_chunk == chunk_size))
```

Comment:
Here it is better to use numpy's builtin assert methods which operate directly on arrays. In this case:
https://docs.scipy.org/doc/numpy-1.15.0/reference/generated/numpy.testing.assert_array_equal.html

Reply:
Done and understood.